src/components/AppLayout/InvalidMasterCopyError/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { Link } from '@gnosis.pm/safe-react-components'
+import { useSelector } from 'react-redux'
+import { getChainInfo } from 'src/config'
+import { Errors, logError } from 'src/logic/exceptions/CodedException'
+import useAsync from 'src/logic/hooks/useAsync'
+import { currentSafe } from 'src/logic/safe/store/selectors'
+import { isValidMasterCopy } from 'src/logic/safe/utils/safeVersion'
+import MuiAlert from '@material-ui/lab/Alert'
+
+const CLI_LINK = 'https://github.com/5afe/safe-cli'
+
+export const InvalidMasterCopyError = ({ onClose }: { onClose: () => void }): React.ReactElement | null => {
+  const chainInfo = getChainInfo()
+  const { implementation } = useSelector(currentSafe)
+
+  const [validMasterCopy, error] = useAsync(async () => {
+    if (implementation.value) {
+      return await isValidMasterCopy(chainInfo.chainId, implementation.value)
```

Comment:
This will run every time the Safe opens. Do we want to cache whether it's valid perhaps?

Reply:
It should only run if the chainId changes / masterCopy address of the SafeInfo changes.
So this should only be after the initial render.

We would prefer to leave it as is for now.

Reply:
Sure but if we have already validated it the once, why need we do so again?

Reply:
The problem here was that if the safe has trouble loading initially we don't have any `implementation.value` and cannot check the validity of the masterCopy. That's why we have to check again when this value changes. 
This caused the bug described by franco in his last comment.

I think its out of scope to optimize this as it should be quite cheap to check the masterCopy and we would have to cache per safe and per chainID. Also backend could decide to add indexing for more masterCopies at some point rendering the cached value invalid.